packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -87,93 +99,143 @@ impl Abigen {
+                #![allow(dead_code)]
+                #![allow(unused_imports)]
 
-                        pub fn get_contract_id(&self) -> &Bech32ContractId {
-                            &self.contract_id
-                        }
+                #includes
 
-                        pub fn get_wallet(&self) -> WalletUnlocked {
-                            self.wallet.clone()
-                        }
+                #code
 
-                        pub fn with_wallet(&self, mut wallet: WalletUnlocked) -> Result<Self, SDKError> {
-                           let provider = self.wallet.get_provider()?;
-                           wallet.set_provider(provider.clone());
+                #abi_structs
+                #abi_enums
 
-                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet, logs_lookup: self.logs_lookup.clone() })
-                        }
+            }
+        })
+    }
 
-                        pub fn logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
-                            extract_and_parse_logs(&self.logs_lookup, receipts)
-                        }
+    fn includes(&self) -> TokenStream {
+        if self.no_std {
+            quote! {
+                use alloc::{vec, vec::Vec};
+                use fuels_core::{EnumSelector, Parameterize, Tokenizable, Token, Identity, try_from_bytes};
+                use fuels_core::types::*;
+                use fuels_core::code_gen::function_selector::resolve_fn_selector;
+                use fuels_types::errors::Error as SDKError;
+                use fuels_types::param_types::ParamType;
+                use fuels_types::enum_variants::EnumVariants;
+            }
+        } else {
+            quote! {
+                use fuels::contract::contract::{Contract, ContractCallHandler};
+                 use fuels::core::{EnumSelector, StringToken, Parameterize, Tokenizable, Token,
+                                  Identity, try_from_bytes};
+                use fuels::core::code_gen::{extract_and_parse_logs, extract_log_ids_and_data};
+                use fuels::core::abi_decoder::ABIDecoder;
+                use fuels::core::code_gen::function_selector::resolve_fn_selector;
+                use fuels::core::types::*;
+                use fuels::signers::WalletUnlocked;
+                use fuels::tx::{ContractId, Address, Receipt};
+                use fuels::types::bech32::Bech32ContractId;
+                use fuels::types::ResolvedLog;
+                use fuels::types::errors::Error as SDKError;
+                use fuels::types::param_types::ParamType;
+                use fuels::types::enum_variants::EnumVariants;
+                use std::str::FromStr;
+                use std::collections::{HashSet, HashMap};
+            }
+        }
+    }
 
-                        #fetch_logs
+    pub fn expand_script(&self) -> Result<TokenStream, Error> {
+        let name = &self.contract_name;
```

Comment:
Setting the script name as `contract_name` could confuse new readers. If that's unavoidable, we should add a comment explaining why we're grabbing the `contract_name` here.